common/src/main/java/org/jboss/pnc/common/monitor/PullingMonitor.java

```diff
@@ -102,6 +102,16 @@ public void monitor(Runnable onMonitorComplete, Consumer<Exception> onMonitorErr
         ObjectWrapper<RunningTask> runningTaskReference = new ObjectWrapper<>();
         Runnable monitor = () -> {
             RunningTask runningTask = runningTaskReference.get();
+
+            if (runningTask == null) {
+                /* There might be a situation where this runnable is called before runningTaskReference is set with the
+                 * running task since this runnable is scheduled to run before runningTaskReference is set. In that
+                 * case, we just skip this runnable and re-run on the next scheduled interval with the assumption that
+                 * runningTaskReference will be set before the next re-run
+                 */
+                log.info("runningTask not set yet inside the 'monitor' runnable! Skipping!");
+                return;
+            }
             try {
                 int waiting = timeWaiting.addAndGet(checkInterval);
```

Comment:
This seems to be never used and can be removed.

Reply:
+1